Orange/widgets/widget.py

```diff
@@ -151,6 +151,11 @@ def __init__(self, *args, **kwargs):
         """QDialog __init__ was already called in __new__,
         please do not call it here."""
 
+        # Test input signal handlers with None == catch deleting connection bugs early
+        for input in self.inputs:
+            args = (None,) if input.single else (None, 0)
+            getattr(self, input.handler)(*args)
+
     @classmethod
```

Comment:
No. This is most definitely not the right place to check this.
At the point when most widgets call the base `__init__` they are not even fully initialized.
e.g.

    inputs = [("Data", int, "set_number")]
    def __init__(self):
        super().__init__()  # will call set_number(None)
        self.number  = None  # member instances are usually defined after
    
    def set_number(self, data):
        if self.number == number:  # BAM: Attribute error
             ...